static/src/javascripts/projects/common/modules/navigation/sticky.js

```diff
@@ -0,0 +1,261 @@
+        mediator.on('modules:nav:open', function () {
+            this.showStickyNavigation();
+        }.bind(this));
+    };
+
+    StickyHeader.prototype.showStickyNavigation = function () {
+        var height = window.innerHeight - $('.js-global-navigation').offset().top;
+
+        $('.js-global-navigation')
+            .addClass('navigation__expandable--sticky')
+            .css('height', height);
+    };
+
+    StickyHeader.prototype.setScrollDirection = function (scrollY) {
+        this.config.direction = (scrollY > this.config.prevScroll) ? 'down' : 'up';
+        this.config.prevScroll = scrollY;
+    };
+
+    StickyHeader.prototype.shouldShowNavigation = function (scrollY) {
+        if (this.config.direction === 'up' && this.config.distance === 0) {
+            this.config.distance = scrollY;
+        }
+
+        // If distance scrolled is more than showNavigationDepth show navigation
+        this.config.showNavigation = (Math.abs(scrollY - this.config.distance) > this.config.showNavigationDepth);
+    };
+
+    StickyHeader.prototype.showNavigation = function (scrollY) {
+        this.shouldShowNavigation(scrollY);
+
+        // If user is scrolling up and navigation threshold was met show navigation
+        if (this.config.direction === 'up' && this.config.showNavigation) {
+            if (this.isTablet || this.isMobile) {
+                this.$els.navigation.removeClass('animate-down-mobile').addClass('animate-up-mobile');
+            } else {
+                this.$els.navigation.removeClass('animate-down-desktop').addClass('animate-up-desktop');
+            }
+
+        } else {
+            // If user is scrolling down and navigation is visible reset bounce distance
+            if (this.config.showNavigation) {
+                // Reset distance bouncing
+                this.config.distance = 0;
+
+                // Close meganav if it's open
+                if (this.$els.navHeader.hasClass('navigation-container--expanded')) {
+                    bean.fire(qwery('.js-navigation-toggle')[0], 'click');
+                }
+            }
+
+            if (this.isTablet || this.isMobile) {
+                this.$els.navigation.removeClass('animate-up-mobile').addClass('animate-down-mobile');
+            } else {
+                this.$els.navigation.removeClass('animate-up-desktop').addClass('animate-down-desktop');
+            }
+        }
+    };
+
+    StickyHeader.prototype.setNavigationDefault = function () {
+        // Make sure navigation is visible and has its default styles
+        this.$els.navigation.removeAttr('class');
+        this.$els.navigation.attr('class', this.navigationClassList);
+    };
+
+    StickyHeader.prototype.updatePosition = function (breakpoint) {
+        var bannerHeight = this.$els.bannerDesktop.dim().height || 128,
+            scrollY;
+
+        fastdom.read(function () {
+            scrollY = $(window).scrollTop();
+        });
+
+        fastdom.write(function () {
```

Comment:
much better to `write` only actual layout-triggering stuff, leave as many calcs out as possible